prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -693,29 +693,51 @@ def run_retrieval(  # type:ignore
     retrieval_type: str,
     metadata: dict[str, Any],
 ) -> tuple[str, str]:
+    context: str = ""
     prompt = output[PSKeys.PROMPTX]
     if retrieval_type == PSKeys.SUBQUESTION:
-        subq_prompt = (
-            f"Generate a sub-question from the following verbose prompt that will"
-            f" help extract relevant documents from a vector store:\n\n{prompt}"
+        subq_prompt: str = (
+            f"I am sending you a verbose prompt \n \n Prompt : {prompt} \n \n"
+            "Generate set of specific subquestions "
+            "from the prompt which can be used to retrive "
+            "relevant context from vector db. "
+            "Ensure that each subquestion is distinct and relevant"
+            "to the different facets of the original query. Do not"
+            "add subquestions for details not mentioned in the original "
+            "prompt. The goal is to maximize retrieval accuracy"
+            " using these subquestions. Please note that, there are cases where the "
+            "response might have a list of answers. The subquestions must not miss out "
+            "any values in these cases. "
+            "Output should be a list of comma seperated "
+            "subquestion prompts. Do not change this format. \n \n "
+            " Subquestions : "
         )
-        prompt = run_completion(
+        subquestions = run_completion(
             llm=llm,
             prompt=subq_prompt,
         )
-    context = _retrieve_context(output, doc_id, vector_index, prompt)
-
-    if not context:
-        # UN-1288 For Pinecone, we are seeing an inconsistent case where
-        # query with doc_id fails even though indexing just happened.
-        # This causes the following retrieve to return no text.
-        # To rule out any lag on the Pinecone vector DB write,
-        # the following sleep is added
-        # Note: This will not fix the issue. Since this issue is inconsistent
-        # and not reproducible easily, this is just a safety net.
-        time.sleep(2)
+        subquestion_list = subquestions.split(",")
+        for each_subq in subquestion_list:
+            retrived_context = _retrieve_context(
```

Comment:
@harini-venkataraman Minor typo in variable name.

Reply:
Thanks @hari-kuriakose, addressed.